toolchain/check/testdata/generic/complete_type.carbon

```diff
@@ -0,0 +1,259 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/generic/complete_type.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/generic/complete_type.carbon
+
+// --- fail_incomplete_in_class.carbon
+
+library "[[@TEST_NAME]]";
+
+class A(T:! type) {
+  // CHECK:STDERR: fail_incomplete_in_class.carbon:[[@LINE+3]]:10: error: `T` evaluates to incomplete type `B` [IncompleteTypeInMonomorphization]
+  // CHECK:STDERR:   var v: T;
+  // CHECK:STDERR:          ^
+  var v: T;
+}
+
+// CHECK:STDERR: fail_incomplete_in_class.carbon:[[@LINE+4]]:1: note: class was forward declared here [ClassForwardDeclaredHere]
+// CHECK:STDERR: class B;
+// CHECK:STDERR: ^~~~~~~~
+// CHECK:STDERR:
+class B;
+
+// CHECK:STDERR: fail_incomplete_in_class.carbon:[[@LINE+6]]:6: error: parameter has incomplete type `A(B)` in function definition [IncompleteTypeInFunctionParam]
+// CHECK:STDERR: fn F(x: A(B)) {}
+// CHECK:STDERR:      ^~~~~~~
+// CHECK:STDERR: fail_incomplete_in_class.carbon:[[@LINE-5]]:1: note: class was forward declared here [ClassForwardDeclaredHere]
+// CHECK:STDERR: class B;
+// CHECK:STDERR: ^~~~~~~~
+fn F(x: A(B)) {}
+
+class B {}
+
+// --- incomplete_in_function.carbon
+
+library "[[@TEST_NAME]]";
+
+fn F(T:! type) {
+  var v: T;
+}
+
+class B;
+
+// F(B) isn't resolved until the end of the file.
```

Comment:
Do we have a test where the deferred resolution discovers an incomplete call in the function body? Is it worth it for contrast?

Reply:
Good idea, I've added one.